src/ImageSharp/Formats/Webp/Lossy/Vp8Encoding.cs

```diff
@@ -68,12 +82,197 @@ static Vp8Encoding()
             }
         }
 
+        // Transforms (Paragraph 14.4)
+        // Does one or two inverse transforms.
         public static void ITransform(Span<byte> reference, Span<short> input, Span<byte> dst, bool doTwo, Span<int> scratch)
         {
-            ITransformOne(reference, input, dst, scratch);
-            if (doTwo)
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Sse2.IsSupported)
+            {
+                // This implementation makes use of 16-bit fixed point versions of two
+                // multiply constants:
+                //    K1 = sqrt(2) * cos (pi/8) ~= 85627 / 2^16
+                //    K2 = sqrt(2) * sin (pi/8) ~= 35468 / 2^16
+                //
+                // To be able to use signed 16-bit integers, we use the following trick to
+                // have constants within range:
+                // - Associated constants are obtained by subtracting the 16-bit fixed point
+                //   version of one:
+                //      k = K - (1 << 16)  =>  K = k + (1 << 16)
+                //      K1 = 85267  =>  k1 =  20091
+                //      K2 = 35468  =>  k2 = -30068
+                // - The multiplication of a variable by a constant become the sum of the
+                //   variable and the multiplication of that variable by the associated
+                //   constant:
+                //      (x * K) >> 16 = (x * (k + (1 << 16))) >> 16 = ((x * k ) >> 16) + x
+
+                // Load and concatenate the transform coefficients (we'll do two inverse
+                // transforms in parallel). In the case of only one inverse transform, the
+                // second half of the vectors will just contain random value we'll never
+                // use nor store.
+                ref short inputRef = ref MemoryMarshal.GetReference(input);
+                var in0 = Vector128.Create(Unsafe.As<short, long>(ref inputRef), 0);
+                var in1 = Vector128.Create(Unsafe.As<short, long>(ref Unsafe.Add(ref inputRef, 4)), 0);
+                var in2 = Vector128.Create(Unsafe.As<short, long>(ref Unsafe.Add(ref inputRef, 8)), 0);
+                var in3 = Vector128.Create(Unsafe.As<short, long>(ref Unsafe.Add(ref inputRef, 12)), 0);
+
+                // a00 a10 a20 a30   x x x x
+                // a01 a11 a21 a31   x x x x
+                // a02 a12 a22 a32   x x x x
+                // a03 a13 a23 a33   x x x x
+                if (doTwo)
```

Comment:
It's better to avoid giving unnecessary extra work to the branch predictor, and refactor `ITransform` into two separate methods.

One hidden cost is that assembly code having branches on a runtime constant will be longer, and needs more cache lines, which are expensive to load.

Alternative "fun" solution, sort of template metaprogramming in C#, that will eliminate branches at JIT time:

```C#
ITransform<TDoTwo>(...) {
    ...
    if (typeof(TDoTwo) == typeof(DoTwo_True)) {
        ...
    }
    else {
       ...
    }
}
```

Can't decide if it's worth it or better to just duplicate the code, and maybe create reusable helper methods for shared bits.